jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -809,6 +817,7 @@ MethodHandle handleForMHInvokeMethods(Class<?> clazz, String methodName, MethodT
 			 * the resulting method handle should be equivalent to one produced by varHandleInvoker().
 			 */
 			if (VarHandle.class.isAssignableFrom(clazz)) {
+				accessCheckArgRetTypes(type);
 				for (AccessMode m : AccessMode.values()) {
 					if (m.methodName().equals(methodName)) {
```

Comment:
Check should be here - in the success case - not at the top of the if scope

Reply:
Yeah, that's right. Fixed, thanks Dan.